dev-tools/mage/integtest_docker.go

```diff
@@ -160,6 +166,38 @@ func (d *DockerIntegrationTester) Test(_ string, mageTarget string, env map[stri
 	return testErr
 }
 
+func saveDockerComposeLogs(rootDir string, mageTarget string, composeEnv map[string]string) error {
+	var (
+		composeLogDir      = filepath.Join(rootDir, "build", "system-tests", "docker-logs")
+		composeLogFileName = filepath.Join(composeLogDir, "TEST-docker-compose-"+mageTarget+".log")
+	)
+
+	if err := os.MkdirAll(composeLogDir, os.ModeDir|os.ModePerm); err != nil {
+		return fmt.Errorf("creating docker log dir: %w", err)
+	}
+
+	composeLogFile, err := os.Create(composeLogFileName)
+	if err != nil {
+		return fmt.Errorf("creating docker log file: %w", err)
+	}
+	defer composeLogFile.Close()
+
+	_, err = sh.Exec(
+		composeEnv,
+		composeLogFile,
```

Comment:
Is the composeLogFile entry needed twice?

Reply:
These two arguments are the stdout and stderr destinations, so if you want to capture everything they are needed. I'll add some inline comments to make this clearer, it does read oddly without them if you don't know what the arguments are.

```go
	_, err = sh.Exec(
		composeEnv,
		composeLogFile, // stdout
		composeLogFile, // stderr
```

Reply:
Got it, thx.